File: Python/python_almost_a_circle_project/models/rectangle.py
"""class rectangle
"""

from models.base import Base

class Rectangle(Base):
    def __init__(self, width, height, x=0, y=0, id=None):
        super().__init__(id)
        self.width = width
        self.height = height
        self.x = x
        self.y = y
    

    def validator(self, name, value, leq):
        """Validator method"""
        if not isinstance(value, int):
           raise TypeError("{} must be an integer".format(name))
        if leq:
           if value <= 0:
               raise ValueError("{} must be > 0".format(name))
        else:
            if value < 0:
                raise ValueError("{} must be >= 0".format(name))
    def area(self):
        """returns the area of a rectangle"""
        return self.__width * self.__height

    def display(self):
        """display the rectangle"""
        for i in range(self.__y):
            for j in range(self.__x):
                print("#", end="")
            print()

    @property
    def width(self):
        """ width getter"""
        return self.__width

    @width.setter
    def width(self, width):
        self.validator("width", width, True)
        self.__width = width

    @property
    def height(self):
        """ width getter"""
        return self.__height

    @height.setter
    def height(self, height):
        self.validator("height", height, True)
        self.__height = height

    @property
    def x(self):
        """ width getter"""
        return self.__x

    @x.setter
    def x(self, x):
        self.validator("x", x, False)
        self.__x = x

    @property
    def y(self):
        """ width getter"""
        return self.__y

    @y.setter
    def y(self, y):
        self.validator("y", y, False)
        self.__y = y

File: JAVA/Assignment4/Rectangle.java
public class Rectangle {
    
    private double width;
    private double height;

    // construtor
    public void setRectangle(double width, double height)
    {
        this.height = height;
        this.width = width;
    }

    // width getter
    public double getWidth()
    {
        return width;
    } 
    // width setter
    public void setWidth(double width){
        this.width = width;
    }
    // height getter
    public double getheight()
    {
        return width;
    } 
    // width setter
    public void setHeight(double height){
        this.height = height;
    }

    public double are(){
        return width * height;
    }
    public static void main(String[] args) {
        Rectangle r = new Rectangle();

        r.height = 5;
        r.width = 4;
        System.out.println("the area of the first rectangle is "+r.are());
        
        Rectangle r2 = new Rectangle();
        r2.height = 8;
        r2.width=  5;
        
        System.out.println("The area of the second rectangle is "+r2.are());
    }

}
